src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -119,7 +122,7 @@ public static class TagHelperBlockRewriter
                 else if (name == null && symbol.Type == HtmlSymbolType.Text)
                 {
                     name = symbol.Content;
```

Comment:
this method is complex enough to need comments at the top of almost every block explaining the current (or at least expected) situation.  e.g. `// Found the start of the name,` here